### PR TITLE
Improve location handling

### PR DESCRIPTION
**Pull request by cbalster**
- also use network provider
- filter new locations by quality before using them
- only submit "precise" locations to server
- some cleanup and refactoring

All the parameters used to filter the location and the definition of what we consider precise should be okay for a start. I'd appreciate it if you could take a look and do a quick test yourselves to verify and maybe correct me ;)

### CODE REVIEW

**Comment by ligi** [comment]
Looks good on a first glance - but want to give it a bit more attention before merging

File: app/src/main/java/de/stephanlindauer/criticalmaps/model/OwnLocationModel.java [hunk 1]
```diff
@@ -22,6 +25,16 @@ public static OwnLocationModel getInstance() {
         return OwnLocationModel.instance;
     }
 
+    public void setLocation(@NonNull GeoPoint location, float accuracy) {
+        ownLocation = location;
+        isLocationPrecise = (accuracy < ACCURACY_PRECISE_THRESHOLD);
+    }
+
+    public boolean hasPreciseLocation() {
+        return isLocationPrecise && (ownLocation != null);
```
Comment (ligi) [inline app/src/main/java/de/stephanlindauer/criticalmaps/model/OwnLocationModel.java]: rlly nit-picky: would turn that around from: is our lo

Comment (cbalster) [inline app/src/main/java/de/stephanlindauer/criticalmaps/model/OwnLocationModel.java]: ha, I actually agree - changed in new push

**Comment by ligi** [comment]
I am for merging this but would be great to merge it into development branch for beta-testing in play - @stephanlindauer has set up open-beta on play

**Comment by stephanlindauer** [comment]
image

**Comment by stephanlindauer** [comment]
development branch is created. i'd suggest that we stick to the git flow rules when doing things on those different branches. http://nvie.com/posts/a-successful-git-branching-model/

**Comment by ligi** [comment]
:+1:  @cbalster can you redirect your PR

**Comment by cbalster** [comment]
replaced by #123

**Pull request closed without merging**